### PR TITLE
Tell curl to not use .curlrc

### PR DESCRIPTION
**Pull request by stefansundin**
Searching the issues and PRs, I wasn't able to find prior discussion of this. Apologies if this is something that has already been decided that it can't be used for some reason.

I have known that Homebrew does not like `~/.curlrc` because `brew doctor` warns about it. But it hasn't been an issue for me until I tried creating a cask (https://github.com/caskroom/homebrew-cask/pull/45612).

In deciding between `-q` and `--disable`, I chose `-q` because it appears that `--disable` was only added in [7.49.0 (May 18 2016)](https://curl.haxx.se/changes.html). I wasn't able to determine when `-q` was initially added.

One reason I can think about against not merging this, is if a user puts special proxy settings in their curlrc. But even in that case I'm not sure, because the user might not expect Homebrew to use that. Is there a way to tell curl to use the macos proxy settings?

Let me know what you think! Thanks!

---

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

### CODE REVIEW

File: Library/Homebrew/test/utils/curl_spec.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+require "utils/curl"
+
+describe "curl" do
+  describe "curl_args" do
+    it "returns -q as the first argument" do
+      expect(curl_args("foo")[1]).to eq("-q")
```
Comment (reitermarkus) [inline Library/Homebrew/test/utils/curl_spec.rb]: `expect(curl_args("foo")).to include("-q")`

Comment (reitermarkus) [reply Library/Homebrew/test/utils/curl_spec.rb]: Ah, ok, fine as is then.

**Comment by MikeMcQuaid** [comment]
Thanks for the PR!

> One reason I can think about against not merging this, is if a user puts special proxy settings in their curlrc.

Perhaps we could have a `HOMEBREW_CURLRC` variable or similar that tells Homebrew (not by default) to use the system `.curlrc`? We could document that under both the Environment Variables and Proxy settings of the manpage.

CC @ilovezfs for thoughts on this PR and the above given we've discussed this before.

File: Library/Homebrew/test/utils/curl_spec.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+require "utils/curl"
+
+describe "curl" do
+  describe "curl_args" do
+    it "returns -q as the first argument" do
+      # -q must be the first argument, or else it does not work
```
Comment (reitermarkus) [inline Library/Homebrew/test/utils/curl_spec.rb]: Replace “or else it does not work” with ”according to `man curl`”.

**Comment by stefansundin** [comment]
Let me know if you want me to remove `check_user_curlrc` from `diagnostics.rb`.

File: Library/Homebrew/utils/curl.rb [hunk 3]
```diff
@@ -12,11 +12,14 @@ def curl_executable
...
 def curl_args(*extra_args, show_output: false, user_agent: :default)
-  args = [
-    curl_executable.to_s,
-    "-q", # do not load .curlrc (must be the first argument)
-    "--show-error",
-  ]
+  args = [curl_executable.to_s]
+
+  # do not load .curlrc unless requested (must be the first argument)
+  if ENV["HOMEBREW_CURLRC"]
```
Comment (stefansundin) [inline Library/Homebrew/utils/curl.rb]: Isn't this backwards? Now we will still load curlrc by default.

Comment (MikeMcQuaid) [reply Library/Homebrew/utils/curl.rb]: @stefansundin It is indeed, I'm a moron! Fixed in https://github.com/Homebrew/brew/pull/4045, thanks for noticing!

**Pull request merged**